### PR TITLE
Improvements for the Continuous Integration with OBS and GitHub/GitLab

### PR DESCRIPTION
**Pull request by danidoni**
* Goal: Help users of OBS get up to speed with the changes introduced since last deliverable.
* Outline of topics / Write what you know:
	- We now detect updates on a PR/MR. We retrigger the workflow not only when the PR/MR is created but also when we detect changes.
	- Until now we reported the latest build result of a package. we now report build results for each package + repository + architecture combination. We also report about multibuild flavours (FIXME: Not sure).
	- We only support triggering workflows from the public instances of github.com and gitlab.com.

### CODE REVIEW

**Comment by hennevogel** [comment]
:heavy_check_mark: Deploy Preview for *openbuildservice* ready!


:hammer: Explore the source changes: 96f82b36b09c729f343eb9338157c727580c2202

:mag: Inspect the deploy log: [https://app.netlify.com/sites/openbuildservice/deploys/60c0d1432750200008f4dff6](https://app.netlify.com/sites/openbuildservice/deploys/60c0d1432750200008f4dff6?utm_source=github&utm_campaign=bot_dl)

:sunglasses: Browse the preview: [https://deploy-preview-299--openbuildservice.netlify.app](https://deploy-preview-299--openbuildservice.netlify.app?utm_source=github&utm_campaign=bot_dp)

File: _posts/development/2021-06-09-scm-integration-improvements.md [hunk 1]
```diff
@@ -1,18 +1,26 @@
-* We now detect updates on a PR/MR. That retriggers the workflow.
-* Until now we reported the latest build result of a package. we now report build results for each package + repository + architecture combination. We also report about multibuild flavours.
-* The format of the workflows file is:
-* We only support triggering workflows from the public instances of github.com and gitlab.com.
+---
+layout: post
+title: Improvements Over Continuous Integration with OBS and GitHub/GitLab
```
Comment (dmarcoux) [inline _posts/development/2021-06-09-scm-integration-improvements.md:3]:
```suggestion
title: Improvements for the Continuous Integration with OBS and GitHub/GitLab
```

File: _posts/development/2021-06-09-scm-integration-improvements.md [hunk 2]
```diff
@@ -1,18 +1,26 @@
...
 
-## Build results for repository and architecture and multibuilds
-OBS will report back to your PR/MR the build results for each repository and architecture combination for a package.
-It also supports multibuilds so it will show the build status of all the flavours of a package.
-This will show up in a commit as a check with the build status reported like this:
+OBS will trigger the workflow again when new changes are detected in an opened PR/MR.
+
+This will allow you to rebuild branched packages if you push new changes to a PR/MR.
+
+## Build results for repository and architecture
```
Comment (dmarcoux) [inline _posts/development/2021-06-09-scm-integration-improvements.md:16]: The capitalization is wrong for this title: https://capitalizemytitle.com/

File: _posts/development/2021-06-09-scm-integration-improvements.md [hunk 3]
```diff
@@ -1,18 +1,26 @@
...
-	✓ OBS Workflow: games/ctris - openSUSE_Tumbleweed/x86_64
+Until now, we were showing only one build result. Now OBS reports the build results back to your PR/MR for each repository and architecture combination of any package.
+
+It will appear in the PR/MR checks section as another check, with the build status reported like this:
+
+	✓ OBS: ctris - openSUSE_Tumbleweed/x86_64
 	
-## SCM Instance support
-Right now we only support react to events coming from github.com or gitlab.com.
+## Major SCM Vendors support
```
Comment (dmarcoux) [inline _posts/development/2021-06-09-scm-integration-improvements.md:24]: Capitalization issues here too: https://capitalizemytitle.com/

File: _posts/development/2021-06-09-scm-integration-improvements.md [hunk 4]
```diff
@@ -1,18 +1,26 @@
...
+
+Right now we only act on events coming from [GitHub.com](https://github.com) or [GitLab.com](https://gitlab.com).
```
Comment (dmarcoux) [inline _posts/development/2021-06-09-scm-integration-improvements.md:26]:
```suggestion
Right now, we only act on events coming from [GitHub.com](https://github.com) or [GitLab.com](https://gitlab.com).
```

File: _posts/development/2021-06-09-scm-integration-improvements.md [hunk 5]
```diff
@@ -1,18 +1,26 @@
...
-It also supports multibuilds so it will show the build status of all the flavours of a package.
-This will show up in a commit as a check with the build status reported like this:
+OBS will trigger the workflow again when new changes are detected in an opened PR/MR.
+
+This will allow you to rebuild branched packages if you push new changes to a PR/MR.
+
+## Build results for repository and architecture
 
-	✓ OBS Workflow: games/ctris - openSUSE_Tumbleweed/x86_64
+Until now, we were showing only one build result. Now OBS reports the build results back to your PR/MR for each repository and architecture combination of any package.
```
Comment (dmarcoux) [inline _posts/development/2021-06-09-scm-integration-improvements.md:18]: This paragraph isn't true.

We reported all build results, but only the last one was displayed since we only had one status for all build results. We kept overwriting the status check. There are now one status check per build result, so this is the major difference here.

File: _posts/development/2021-06-09-scm-integration-improvements.md (hunk 2 above)
Comment (danidoni) [reply _posts/development/2021-06-09-scm-integration-improvements.md]: Fixed

File: _posts/development/2021-06-09-scm-integration-improvements.md (hunk 3 above)
Comment (danidoni) [reply _posts/development/2021-06-09-scm-integration-improvements.md]: Fixed

File: _posts/development/2021-06-09-scm-integration-improvements.md (hunk 5 above)
Comment (danidoni) [reply _posts/development/2021-06-09-scm-integration-improvements.md]: Okay, what about [this](https://github.com/openSUSE/obs-landing/pull/299/files#diff-e75abc542fb8f3e86a0045bfbf08b9b9d6843c855dda1bc739e465717a98a84cR20)

Comment (dmarcoux) [reply _posts/development/2021-06-09-scm-integration-improvements.md]: This is fine, but the sentence above is still somehow confusing/wrong. This is what I would explain, maybe rephrase it if needed.

We reported all build results, but only the last one was displayed since we only had one status for all build results. We kept overwriting the status check.

File: _posts/development/2021-06-09-scm-integration-improvements.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
+---
+layout: post
+title: Improvements for the Continuous Integration with OBS and GitHub/GitLab
+category: development
+---
+## Introduction
+
+Today we have some improvements over our [first attempt at Continuous Integration](https://openbuildservice.org/2021/05/31/scm-integration/).
```
Comment (dmarcoux) [inline _posts/development/2021-06-09-scm-integration-improvements.md:8]:
```suggestion
Today, we have some improvements for our [first iteration at continuous integration](https://openbuildservice.org/2021/05/31/scm-integration/).
```

File: _posts/development/2021-06-09-scm-integration-improvements.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
+---
+layout: post
+title: Improvements for the Continuous Integration with OBS and GitHub/GitLab
+category: development
+---
+## Introduction
```
Comment (dmarcoux) [inline _posts/development/2021-06-09-scm-integration-improvements.md:6]: This header should be removed.

```suggestion
```

File: _posts/development/2021-06-09-scm-integration-improvements.md (hunk 5 above)
Comment (danidoni) [reply _posts/development/2021-06-09-scm-integration-improvements.md]: Okay, I've reworded it a bit.

File: _posts/development/2021-06-09-scm-integration-improvements.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
+---
+layout: post
+title: Improvements for the Continuous Integration with OBS and GitHub/GitLab
+category: development
+---
+
+Today, we have some improvements for our [first iteration at continuous integration](https://openbuildservice.org/2021/05/31/scm-integration/).
```
Comment (hennevogel) [inline _posts/development/2021-06-09-scm-integration-improvements.md:7]: The meaning of `iteration` is not what you think for your readers. This is something that makes sense for us in the team, not for anyone else.

File: _posts/development/2021-06-09-scm-integration-improvements.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+## Updates on a PR/MR
+
+OBS will trigger the workflow again when new changes are detected in an opened PR/MR.
+
+This will allow you, for example, to rebuild branched packages when pushing new changes to the PR/MR.
```
Comment (hennevogel) [inline _posts/development/2021-06-09-scm-integration-improvements.md:13]: You make it sound as if there is more. There isn't right?

File: _posts/development/2021-06-09-scm-integration-improvements.md [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+## Build Results for Repository and Architecture
+
+You maybe had an integration set for a package built against several distributions and architectures. OBS was reporting all build results, but we were showing only the last one on the checks section of a PR/MR. Now OBS shows the build results for each repository and architecture combination of any package.
```
Comment (hennevogel) [inline _posts/development/2021-06-09-scm-integration-improvements.md:17]: integration? What's that? You mean workflow?

File: _posts/development/2021-06-09-scm-integration-improvements.md [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+
+	✓ OBS: ctris - openSUSE_Tumbleweed/x86_64
+	
+If your package build several flavours, the check will have the flavour name appended to the package name:
+
+	✓ OBS: ctris:hans - openSUSE_Tumbleweed/x86_64
+	
+TODO: Add screenshots of an actual checks section
+
+## Major SCM Vendors Support
```
Comment (hennevogel) [inline _posts/development/2021-06-09-scm-integration-improvements.md:29]: You make this sound as if it is a feature. But it's a limitation right? :)

**Review by hennevogel — CHANGES_REQUESTED** [review]

File: _posts/development/2021-06-09-scm-integration-improvements.md (hunk 8 above)
Comment (dmarcoux) [reply _posts/development/2021-06-09-scm-integration-improvements.md:7]: I disagree with you on this one. Looking at the definition of `iteration`:

> Iteration is the repetition of a process in order to generate an outcome. The sequence will approach some end point or end value. Each repetition of the process is a single iteration, and the outcome of each iteration is then the starting point of the next iteration.

This is what we're doing. What is it for our readers if it's not that?

Comment (danidoni) [reply _posts/development/2021-06-09-scm-integration-improvements.md]: I understand `iteration` is internal to our team's working methodology, that our users don't really care. Nevertheless I've changed that sentence so...

**Comment by danidoni** [comment]
@hennevogel @dmarcoux Review again, please

File: _posts/development/2021-06-09-scm-integration-improvements.md [hunk 12]
```diff
@@ -0,0 +1,31 @@
+---
+layout: post
+title: Improvements for the Continuous Integration with OBS and GitHub/GitLab
+category: development
+---
+
+Today, we have some improvements in the Continuous Integration we unveiled in our [last blog post.](https://openbuildservice.org/2021/05/31/scm-integration/).
```
Comment (dmarcoux) [inline _posts/development/2021-06-09-scm-integration-improvements.md:7]: There is no need to capitalize continuous integration, it's not a proper noun. [I already commented about that earlier](https://github.com/openSUSE/obs-landing/pull/299#discussion_r648976840)

```suggestion
Today, we have some improvements for the continuous integration we unveiled in our [previous blog post](https://openbuildservice.org/2021/05/31/scm-integration/).
```

File: _posts/development/2021-06-09-scm-integration-improvements.md [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+
+Today, we have some improvements for the continuous integration we unveiled in our [previous blog post](https://openbuildservice.org/2021/05/31/scm-integration/).
+
+## Updates on a PR/MR
+
+OBS will trigger the workflow again when new changes are detected in an opened PR/MR. It will rebuild your branched packages when pushing new commits, basically.
+
+## Build Results for Repository and Architecture
+
+Maybe you had a workflow set for a package built against several distributions and architectures. OBS was reporting all build results, but we were showing only the last one on the checks section of a PR/MR. Now OBS shows the build results for each repository and architecture combination of any package.
```
Comment (hennevogel) [inline _posts/development/2021-06-09-scm-integration-improvements.md:15]: I understand what you are trying to say. The OBS shows all build results but the PR only the last one. But this is pretty hard to parse from those sentences. Maybe you can make this a bit simpler?

**Review by hennevogel — CHANGES_REQUESTED** [review]

File: _posts/development/2021-06-09-scm-integration-improvements.md (hunk 13 above)
Comment (danidoni) [reply _posts/development/2021-06-09-scm-integration-improvements.md]: I've reworded that paragraph. Check it out again, please

**Comment by dmarcoux** [comment]
It would have been nice to squash the commits before merging... Anyway, it's too late now.
